--- osc-source_validator/20-files-present-and-referenced
+++ osc-source_validator/20-files-present-and-referenced
@@ -258,21 +258,24 @@
 	    grep -a -x $BASE $TMPDIR/sources > /dev/null && continue
 	    echo Attention, $BASE is not mentioned in spec files as source or patch.
 	    if [ -d $DIR_TO_CHECK/$BASE ] ; then
+                # be a bit more relaxed for osc, it won't upload directories anyway
+                if [ ! -d $DIR_TO_CHECK/.osc ] ; then
 		echo "!! $BASE is a directory !!"
-		if test "$BATCHMODE" != true ; then
-		    echo "remove the whole subtree with 'd'"
-		    echo -n "Is this correct? [N/y/d] (y to ignore) "
-		    read ANSWER
-		    test "$ANSWER" = y -o "$ANSWER" = Y || {
-			if test "$ANSWER" = d -o "$ANSWER" = D ; then
-			    rm -Rfv $DIR_TO_CHECK/$BASE
-			else
-			    echo ok, please fix it...
-			    RETURN=1
-			fi
-		    }
-		else
-		    echo "###ASK -r $DIR_TO_CHECK/$BASE"
+		    if test "$BATCHMODE" != true ; then
+		        echo "remove the whole subtree with 'd'"
+		        echo -n "Is this correct? [N/y/d] (y to ignore) "
+		        read ANSWER
+		        test "$ANSWER" = y -o "$ANSWER" = Y || {
+		    	if test "$ANSWER" = d -o "$ANSWER" = D ; then
+		    	    rm -Rfv $DIR_TO_CHECK/$BASE
+		    	else
+		    	    echo ok, please fix it...
+		    	    RETURN=1
+		    	fi
+		        }
+		    else
+		        echo "###ASK -r $DIR_TO_CHECK/$BASE"
+		    fi
 		fi
 	    else
 		if test "$BATCHMODE" != true ; then
--- osc-source_validator/45-stale-changes
+++ osc-source_validator/45-stale-changes
@@ -13,8 +13,8 @@
 
 for i in $SPECLIST ; do
     test -f $DIR_TO_CHECK/$i.changes || {
-	echo "ERROR: $i.changes does not exist."
-	RETURN=1
+	echo "WARNING: $i.changes does not exist. This package can not be submitted to openSUSE product projects."
+	exit 0
     }
 done
 # check for stale .changes files
